beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -149,31 +145,12 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
     pub fn get_missing_blob_ids(
```

Comment:
this enum works well, i ended up using it in some of the single blob lookup code too. right now, we have simple logic
- if we're prior to deneb, blobs aren't required
- if we see a block, we know everything
- if we don't see a block, we know nothing

but we could update this to 
- if we see a blob at index 4, we know all blobs with index < 4 must exist and blobs with index > 4 may exist

this enum doesn't give us both of these pieces at once, but if we end up wanting that info i think this could be pretty easily extended for that

Reply:
> if we see a blob at index 4, we know all blobs with index < 4 must exist and blobs with index > 4 may exist

isn't this not necessarily true given that the proposer could give us a blob that's not in the block to fake us out?

Reply:
Yes that's a good point so we'd have to make sure that info were updated whenever we do get the block. Might not be worth implementing, I'm not sure it gives us much benefit over just querying for everything